modules/core/shared/src/main/scala/gem/Asterism.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+
+import cats.Eq
+import cats.data.NonEmptyList
+
+import gem.enum.{ AsterismType, Instrument }
+
+import monocle.Prism
+import monocle.macros.{ Lenses, GenPrism }
+
+
+sealed trait Asterism extends Product with Serializable {
+
+  def asterismType: AsterismType
+
+  def instrument: Instrument
```

Comment:
As you hinted, we may want this to be indexed on the instrument; i.e., `Asterism[I <: Instrument with Singleton]` and `val instrument: I`.

Reply:
Or `trait Asterism { type I <: Instrument with Singleton ...` and then `Asterism.Aux`.

Reply:
I've been struggling a bit with this.  Will I need a single-target asterism instance for each instrument that it applies to?

Reply:
`case class Single[I0 <: Instrument with Singleton](...) extends Asterism { type I = I0 }` .. something like that?